src/internal/common/ptr/pointer.go

```diff
@@ -19,3 +21,25 @@ func Val[T any](ptr *T) T {
 
 	return *ptr
 }
+
+// ValOK behaves the same as Val, except it also gives
+// a boolean response for whether the pointer was nil
+// (false) or non-nil (true).
+func ValOK[T any](ptr *T) (T, bool) {
+	if ptr == nil {
+		return *new(T), false
+	}
+
+	return *ptr, true
+}
+
```

Comment:
The wording is a little misleading in comparison with the logic. If it was nil, the first clause should return true and the second one is false. Otherwise, the comments require updating.

Reply:
mm, no, I think it's correct.  Consider this in comparison to indexing map values:
```go
m := map[int]any{}
v, ok := m[key] // ok is true if the key holds an entry in the map.  ie: "its present"

p *int = 0
v, ok := ptr.ValOK(p) // ok is true if p is non-nil.  ie: "its present"
```